ietf/meeting/forms.py

```diff
@@ -187,24 +183,14 @@ def is_virtual(self):
             return True
 
     def set_group_options(self):
-        '''Set group options based on user accessing the form'''
-        if has_role(self.user, "Secretariat"):
-            return  # don't reduce group options
-        q_objects = Q()
-        if has_role(self.user, "Area Director"):
-            q_objects.add(Q(type__in=["wg", "ag", "team"], state__in=("active", "proposed", "bof")), Q.OR)
-        if has_role(self.user, "IRTF Chair"):
-            q_objects.add(Q(type__in=["rg", "rag"], state__in=("active", "proposed")), Q.OR)
-        if has_role(self.user, "WG Chair"):
-            q_objects.add(Q(type="wg", state__in=("active", "proposed", "bof"), role__person=self.person, role__name="chair"), Q.OR)
-        if has_role(self.user, "RG Chair"):
-            q_objects.add(Q(type="rg", state__in=("active", "proposed"), role__person=self.person, role__name="chair"), Q.OR)
-        if has_role(self.user, "Program Lead") or has_role(self.user, "Program Chair"):
-            q_objects.add(Q(type="program", state__in=("active", "proposed"), role__person=self.person, role__name__in=["chair", "lead"]), Q.OR)
-        
-        queryset = Group.objects.filter(q_objects).distinct().order_by('acronym')
+        """Set group options based on user accessing the form"""
+        queryset = groups_managed_by(
+            self.user,
+            Group.objects.with_meetings(),
+        ).filter(
+            state_id__in=['active', 'proposed', 'bof']
```

Comment:
The previous code had based its selections on a cascade of criteria of role, group type, and group state.  It looks to me like the logic here is to consult the database for all of that.  I don't have a copy of the database in front of me, so I have to ask you - how accurately does the database model the previous logic, and the desired logic?

Reply:
The DB is authoritative on the matter and getting rid of the old code here was one of the major goals. It was a rough approximation that worked only because most users fell into the categories that the old cascade handled (aided, at least in testing, by a bug that caused some users to have access to *all* groups, whether or not they should).

The intention is that this method should find all the groups/roles that cause a user to be granted access to the views using this form. (The bug mentioned came up when a user was granted access to the form but the old logic did not return a list that included the group whose role triggered that access.)